.github/workflows/ci-tests.yml

```diff
@@ -0,0 +1,64 @@
+name: Go Test Workflow
+
+on:
+  push:
+    branches: [master]
+  pull_request:
+    branches: [master]
+    
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    
+    steps:
+    - name: Setup Go
+      uses: actions/setup-go@v2
+      with:
+        go-version: '^1.19'
+    
+    - name: Checkout Code
+      uses: actions/checkout@v4
+      with:
+        ref: ${{ github.event.pull_request.head.ref }}
+      
+    - name: Get dependencies
```

Comment:
I don't think this is necessary

Reply:
So far, I have it for:

- Checking that the depencies are available and reachable
- Ensure that the pinned version of each dependency is correct

Could you elaborate a little longer why is not necessary? Do you think that the above points are already covered by any other step? Also, if we do not pull dependencies, will it even run the tests?